### PR TITLE
Prepare major release

### PR DESCRIPTION
**Pull request by core23**
## Subject

Removed deprecated classes. So we have a fresh start for the repo split.

I am targeting this branch, because this is a BC break.

### CODE REVIEW

File: src/CoreBundle/Resources/config/form_types.xml [hunk 1]
```diff
@@ -1,46 +1,46 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <services>
-        <service id="sonata.core.form.type.array_legacy" class="Sonata\CoreBundle\Form\Type\ImmutableArrayType" public="true">
+        <service id="sonata.core.form.type.array_legacy" class="Sonata\Form\Type\ImmutableArrayType" public="true">
```
Comment (OskarStark) [inline src/CoreBundle/Resources/config/form_types.xml]: Why the legacy suffix?

**Comment by kunicmarko20** [comment]
So this is the last release? after this people should depend on split repos?

**Comment by core23** [comment]
The master is the 4.x development branch, so we should remove all deprecated classes

Not sure, if we will ever release a 4.0 release...

**Comment by core23** [comment]
Can you please review and merge @kunicmarko20

**Comment by kunicmarko20** [comment]
Thanks @core23

**Pull request merged**